lib/Sema/CSDiag.cpp

```diff
@@ -2430,6 +2430,12 @@ bool FailureDiagnosis::diagnoseGeneralMemberFailure(Constraint *constraint) {
     }
   }
 
+  // If base type has unresolved generic parameters, such might mean
+  // that it's initializer with erroneous argument, otherwise this would
+  // be a simple ambigious archetype case, neither can be diagnosed here.
+  if (baseTy->hasTypeParameter())
```

Comment:
I think that's actually going to be better, that way we make sure that base is "complete" before doing anything with it, I'll make the change a bit later.